cartographer_rviz/cartographer_rviz/drawable_trajectory.cc

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer_rviz/drawable_trajectory.h"
+
+#include "cartographer/common/make_unique.h"
+
+namespace cartographer_rviz {
+
+DrawableTrajectory::DrawableTrajectory(const int trajectory_id,
+                                       ::rviz::Property* submap_category,
+                                       Ogre::SceneManager* scene_manager,
+                                       const bool visible)
+    : id_(trajectory_id),
+      submaps_category_(submap_category),
+      scene_manager_(scene_manager),
+      submaps_() {
+  property_ = new ::rviz::BoolProperty(
+      QString("Trajectory %1").arg(id_), visible,
+      QString("List of all submaps in Trajectory %1.").arg(id_),
+      submaps_category_, SLOT(AllEnabledToggled()), this);
+}
+
+DrawableTrajectory::~DrawableTrajectory() {
+  submaps_.clear();
```

Comment:
Sounds to me something else is wrong. A map (sorry, not a vector as I stated above) will definitely destroy all the objects contained inside when it's being destroyed.